lib/tree.rb

```diff
@@ -16,47 +16,149 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n)
+  # Space Complexity: O(1)
```

Comment:
Actually since you're doing a recursive solution the system stack will be of size O(log n) if the tree is balanced and O(n) otherwise.

Also your time complexity will be the same.